rustls/tests/api.rs

```diff
@@ -4194,3 +4196,151 @@ fn test_no_warning_logging_during_successful_sessions() {
         });
     }
 }
+
+/// Test that secrets can be extracted.
+#[cfg(feature = "extract_secrets")]
+#[test]
+fn test_extract_secrets() {
+    use ring::aead::LessSafeKey;
+
+    let kt = KeyType::Rsa;
```

Comment:
Just a few comments in this long function would make it easier to understand how this is testing the secret extractions, I think.

Reply:
I did that, added another test, and clarified some of the code here as well.

Reply:
Great, thanks!